src/jsesc.js

```diff
@@ -205,6 +208,9 @@ const jsesc = (argument, options) => {
 			if (useOctNumbers) {
 				return '0o' + argument.toString(8);
 			}
+		} else if (isDate(argument)) {
+			// Date objects have a toJSON method, which implies json: false here.
+			return 'new Date(' + jsesc(argument.valueOf(), options) + ')';
```

Comment:
```suggestion
			return 'new Date(' + jsesc(+argument, options) + ')';
```

Reply:
I'm generally not in favor of the unary plus operator for type coercion, but I'll defer to @mathiasbynens on stylistic points.

Reply:
Is this purely stylistic, @michaelficarra, or are you getting at something else here? (`valueOf` can be overwritten, but it seems like in this case it would still be called no matter what.)

Reply:
Purely stylistic. Unary + is a very common way to get a timestamp from a date.

Reply:
Thanks for clarifying. I weakly prefer the more explicit `.valueOf()` to be honest, so let's just keep this as-is.